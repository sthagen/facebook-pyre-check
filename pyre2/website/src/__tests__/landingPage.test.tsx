/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @format
 */

import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import Home from '../pages/index';

describe('Home page', () => {
    test('renders internal site correctly', () => {
        process.env.INTERNAL_STATIC_DOCS = '1';
        const { container } = render(<Home />);

        expectHeaderAndTitleIsCorrect(container);

        // Verify that the new sections are present
        const whyPyreFlySection = document.getElementById('why-pyrefly-section');
        expect(whyPyreFlySection).toBeInTheDocument();

        const performanceComparisonSection = document.getElementById('performance-comparison-section');
        expect(performanceComparisonSection).toBeInTheDocument();

        const quotesSection = document.getElementById('quotes-section');
        expect(quotesSection).toBeInTheDocument();

        // Run test with --update-snapshot to update the snapshot if the test is failing after
        // you made a intentional change to the home page
        expect(container).toMatchSnapshot();
    });

    test('renders external site correctly', () => {
        process.env.INTERNAL_STATIC_DOCS = undefined;
        const { container } = render(<Home />);

        expectHeaderAndTitleIsCorrect(container);

        // Verify that the new sections are not present
        const whyPyreFlySection = document.getElementById('why-pyrefly-section');
        expect(whyPyreFlySection).not.toBeInTheDocument();

        const performanceComparisonSection = document.getElementById('performance-comparison-section');
        expect(performanceComparisonSection).not.toBeInTheDocument();

        const quotesSection = document.getElementById('quotes-section');
        expect(quotesSection).not.toBeInTheDocument();

        // Run test with --update-snapshot to update the snapshot if the test is failing after
        // you made a intentional change to the home page
        expect(container).toMatchSnapshot();
    });


    function expectHeaderAndTitleIsCorrect(container: HTMLElement) {
        // Verify that title is set correctly
        expect(document.title).toBe('Pyrefly: A Static Type Checker for Python');

        // Verify that fireflies are present
        const fireflyElements = document.querySelectorAll('#firefly');
        expect(fireflyElements.length).toEqual(5);

        // Verify that header is correct
        const header = container.querySelector('header');
        expect(header).toBeInTheDocument();
        const title = header?.querySelector('h1');
        expect(title).toBeInTheDocument();
        expect(title?.textContent).toBe('pyrefly.');

        // verify subtitle is correct
        const subtitle = container.querySelector('header p');
        expect(subtitle).toBeInTheDocument();
        const subtitleText = subtitle?.textContent?.trim();
        expect(subtitleText).toBe('A faster Python type checker written in Rust');

        const comingSoonLink = header?.querySelector('a');
        expect(comingSoonLink).toBeInTheDocument();
        expect(comingSoonLink?.getAttribute('href')).toBe('https://github.com/facebook/pyrefly/milestone/1');
    }
});
