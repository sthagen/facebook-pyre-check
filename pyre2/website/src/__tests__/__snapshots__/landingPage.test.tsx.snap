// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page renders external site correctly 1`] = `
<div>
  <div>
    <div>
      Pyrefly: A Static Type Checker for Python
    </div>
    <div>
      Pyrefly is a static type checker for Python
    </div>
    <div>
      <header>
        <div
          class="container"
        >
          <h1>
            pyrefly
            <span>
              .
            </span>
          </h1>
          <p>
            <span>
              A faster Python type checker written in Rust
            </span>
          </p>
          <section>
            <a
              href="https://github.com/facebook/pyrefly/milestone/1"
            >
               Github 
            </a>
            <a
              href="https://pyrefly.org/try/"
            >
               Demo 
            </a>
            <a
              href="https://pyrefly.org/en/docs/learn-python-typing/"
            >
               Docs 
            </a>
          </section>
          <section>
            <p>
              Launching Spring 2025
            </p>
          </section>
          <section>
            <li
              id="firefly"
            />
            <li
              id="firefly"
            />
            <li
              id="firefly"
            />
            <li
              id="firefly"
            />
            <li
              id="firefly"
            />
          </section>
        </div>
      </header>
    </div>
  </div>
</div>
`;

exports[`Home page renders internal site correctly 1`] = `
<div>
  <div>
    <div>
      Pyrefly: A Static Type Checker for Python
    </div>
    <div>
      Pyrefly is a static type checker for Python
    </div>
    <div>
      <header>
        <div
          class="container"
        >
          <h1>
            pyrefly
            <span>
              .
            </span>
          </h1>
          <p>
            <span>
              A faster Python type checker written in Rust
            </span>
          </p>
          <section>
            <a
              href="https://github.com/facebook/pyrefly/milestone/1"
            >
               Github 
            </a>
            <a
              href="https://pyrefly.org/try/"
            >
               Demo 
            </a>
            <a
              href="https://pyrefly.org/en/docs/learn-python-typing/"
            >
               Docs 
            </a>
          </section>
          <section>
            <p>
              Launching Spring 2025
            </p>
          </section>
          <section>
            <li
              id="firefly"
            />
            <li
              id="firefly"
            />
            <li
              id="firefly"
            />
            <li
              id="firefly"
            />
            <li
              id="firefly"
            />
          </section>
        </div>
      </header>
      <section
        id="why-pyrefly-section"
      >
        <div />
        <div
          class="container"
        >
          <div>
            <div>
              <h3>
                Scale with Confidence
              </h3>
              <p>
                 Typecheck 1.8 million lines of code per second
              </p>
            </div>
            <div>
              <h3>
                Developer Delight
              </h3>
              <p>
                 Catch errors as you write code, and get instant feedback right in your favorite editor
              </p>
            </div>
            <div>
              <h3>
                Easy Onboarding
              </h3>
              <p>
                 Start type checking your code in minutes
              </p>
            </div>
            <div>
              <h3>
                Codebase Clarity
              </h3>
              <p>
                Improve code readability and maintainability with precise type
          annotations
              </p>
            </div>
            <div>
              <h3>
                Tooling for Improving Types
              </h3>
              <p>
                Comes packaged with tools to elevate your type coverage
          from 0 to 100
              </p>
            </div>
            <div>
              <h3>
                Fully Featured
              </h3>
              <p>
                 Actively under development. Follows the typing standards introduced in PEPs 484, 526, 612.
              </p>
            </div>
          </div>
        </div>
      </section>
      <section
        id="performance-comparison-section"
      >
        <div />
        <div />
        <div
          class="container"
        >
          <h2>
            Performance Comparison
          </h2>
          <div>
            <div>
              <button>
                PyTorch
              </button>
              <button>
                Instagram
              </button>
              <button>
                Example
              </button>
            </div>
            <div>
              <div>
                <div>
                  <span>
                    <strong>
                      Pyrefly
                    </strong>
                  </span>
                  <div>
                    <div
                      style="height: 80%; width: 3.3333333333333335%;"
                    >
                      <div
                        style="animation-duration: 0.5s;"
                      />
                    </div>
                  </div>
                  <span>
                    <strong>
                      0s
                    </strong>
                  </span>
                </div>
                <div>
                  <span>
                    <strong>
                      MyPy
                    </strong>
                  </span>
                  <div>
                    <div
                      style="height: 80%; width: 53.333333333333336%;"
                    >
                      <div
                        style="animation-duration: 8s;"
                      />
                    </div>
                  </div>
                  <span>
                    <strong>
                      0s
                    </strong>
                  </span>
                </div>
                <div>
                  <span>
                    <strong>
                      Pyright
                    </strong>
                  </span>
                  <div>
                    <div
                      style="height: 80%; width: 33.33333333333333%;"
                    >
                      <div
                        style="animation-duration: 5s;"
                      />
                    </div>
                  </div>
                  <span>
                    <strong>
                      0s
                    </strong>
                  </span>
                </div>
                <div>
                  <span>
                    <strong>
                      Pytype
                    </strong>
                  </span>
                  <div>
                    <div
                      style="height: 80%; width: 80%;"
                    >
                      <div
                        style="animation-duration: 12s;"
                      />
                    </div>
                  </div>
                  <span>
                    <strong>
                      0s
                    </strong>
                  </span>
                </div>
                <div>
                  <span>
                    <strong>
                      Pyre1
                    </strong>
                  </span>
                  <div>
                    <div
                      style="height: 80%; width: 100%;"
                    >
                      <div
                        style="animation-duration: 15s;"
                      />
                    </div>
                  </div>
                  <span>
                    <strong>
                      0s
                    </strong>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
      <section
        id="quotes-section"
      >
        <div />
        <div
          class="container"
        >
          <h2>
            Don't just take our word for it
          </h2>
          <div>
            <div>
              <p>
                Pyre and Lint are hands down the most important investments Instagram has made into engineering productivity for our Python codebase. We could not have maintained our product dev velocity with so many people working in the codebase, all while serving 2 billion active users, without the safety and security of static analysis.
              </p>
              <div>
                <strong>
                  &lt;TO BE ADDED&gt;
                </strong>
                <span>
                  Instagram Leadership
                </span>
              </div>
            </div>
            <div>
              <p>
                The enhanced type system and clear error messages have made our
        codebase more maintainable and easier to debug.
              </p>
              <div>
                <strong>
                  &lt;TO BE ADDED&gt;
                </strong>
                <span>
                  core Contributor, PyTorch
                </span>
              </div>
            </div>
            <div>
              <p>
                As an early adopter, I'm impressed by Pyrefly's extensibility
        and the active community support. It's the future of Python type
        checking.
              </p>
              <div>
                <strong>
                  &lt;TO BE ADDED&gt;
                </strong>
                <span>
                  &lt;Open Source Project Name - TO BE ADDED&gt;
                </span>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>
</div>
`;
